examples/QIR/Simulation/README.md

```diff
@@ -0,0 +1,193 @@
+Different parts of the QIR spec are then implemented in C++ in the [lib/QIR](https://github.com/microsoft/qsharp-runtime/tree/main/src/Qir/Runtime/lib/QIR) folder.
+For example, the `__quantum__rt__string_create` QIR function is translated to the C++ `quantum__rt__string_create` function which resides in [lib/QIR/strings.cpp](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/lib/QIR/strings.cpp).
+For others such as `__quantum__rt__qubit_allocate`, the Runtime will eventually call the implementation provided by the driver.
+
+Other components that are used by QIR simulators for Q# are provided by [lib/QSharpCore](https://github.com/microsoft/qsharp-runtime/tree/main/src/Qir/Runtime/lib/QSharpCore) and [lib/QSharpFoundation](https://github.com/microsoft/qsharp-runtime/tree/main/src/Qir/Runtime/lib/QSharpFoundation).
+In particular, the `QSharpCore` component provides the Q# instruction set.
+Similar to above, a bridge [lib/QSharpCore/qsharp-core-qis.ll](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll) first translates a function such as `__quantum__qis__h__body` to `quantum__qis__h__body` located in [lib/QSharpCore/intrinsics.cpp](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/lib/QSharpCore/intrinsics.cpp), which then calls upon the specific implementation provided by the gate set interface.
+
+A qubit manager (see next section) is also provided in [lib/QIR/QubitManager.cpp](https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Runtime/lib/QIR/QubitManager.cpp).
+
+## Qubit Management
+
+Generally, we can distinguish three layers of qubit representation or abstractions.
+Two of those are frequently encountered in quantum computing literature, the *logical qubit* and *the physical* qubit.
+
+A logical qubit is an idealized entity that represents the Qubit from quantum computation theory: a perfect two-level system that behaves as expected when applying quantum gates or measurements.
+The physical qubit on the other hand is a real physical system that *approximately* behaves as a two-level system, and is imperfectly manipulated and isolated from the environment, leading to a limited reliability and information life-time.
+
+There are various techniques to create logical qubits from physical qubits via the use of error correction schemes and fault-tolerant systems.
+Whether the logical qubit is implemented by 1 or 1000 physical qubits, which type of physical system is used, or even whether one logical qubit remains on the same physical qubits throughout the entire computation, is not relevant at the logical level and is thus abstracted away.
+Quantum computing algorithms are typically formulated at the logical level unless they specifically deal with error correction or other aspects of physical qubits.
```

Comment:
I agree that this is definitely true of scalable devices, but may be worth calling out that this is _not_ true of near-term devices in general?